django/db/models/aggregates.py

```diff
@@ -211,3 +213,16 @@ def __init__(self, expression, sample=False, **extra):
 
     def _get_repr_options(self):
         return {**super()._get_repr_options(), "sample": self.function == "VAR_SAMP"}
+
+
+class JSONArrayAgg(Aggregate):
+    function = "JSON_ARRAYAGG"
+    name = "JSONArrayAgg"
+    output_field = JSONField()
+    arity = 1
+
+    def as_sqlite(self, compiler, connection, **extra_context):
+        "Use JSON_GROUP_ARRAY function on SQLite."
+        return super().as_sqlite(
+            compiler, connection, function="JSON_GROUP_ARRAY", **extra_context
+        )
```

Comment:
Don't we also need to use `json_array` on Postgres?

Reply:
Why do we need to use `json_array` when Postgres has added support for [json_array](https://www.postgresql.org/docs/16/functions-json.html) and [json_arrayagg](https://www.postgresql.org/docs/16/functions-aggregate.html) both in version 16? 🤔

Reply:
[Django supports Postgres 13+](https://docs.djangoproject.com/en/5.1/ref/databases/#postgresql-notes) so we [can't use a function only available in Postgres 16+](https://www.postgresql.org/docs/current/release-16.html#RELEASE-16-FUNCTIONS).

I suspect we'll have to emulate `json_arrayagg` using `array_agg` on Postgres < 16.

Reply:
Thanks emulating using `array_agg` makes sense.

Reply:
Hey friends, i would love to have just your thoughts  and feedback on this before i update my PR shortly, so far this is my emulated version of `JSON_ARRAYAGG` using `ARRAY_AGG`.


   
```
    def as_postgresql(self, compiler, connection, **extra_context):
        if not connection.features.is_postgresql_16:
            return super().as_sql(
                compiler,
                connection,
                function="ARRAY_AGG",
                template="ARRAY_TO_JSON(%(function)s(%(expressions)s))::text",
                **extra_context
            )
```